src/lua/lua_cryptobox.c

```diff
@@ -1001,10 +1002,9 @@ rspamd_lua_ssl_hmac_create(struct rspamd_lua_cryptobox_hash *h, const EVP_MD *ht
 	(defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x30500000)
 	h->content.hmac_c = g_malloc0(sizeof(*h->content.hmac_c));
 #else
-	EVP_MAC *mac = EVP_MAC_fetch(libctx, "HMAC", NULL);
+	EVP_MAC *mac = EVP_MAC_fetch(NULL, "HMAC", properties);
```

Comment:
Please note, that these functions are ONLY for OpenSSL 3.0+. We need to be compatible with OpenSSL 1.0 and LibreSSL, so we can't unconditionally use OpenSSL 3.0 functions - it must be guarded with `#ifdef`. And yes, OpenSSL sucks, I know.